WP4/D4.6/report.tex

```diff
@@ -0,0 +1,186 @@
+
+\begin{figure}
+    \center
+    \includegraphics[width=8cm]{img/nbdiff}
+    \caption{nbdiff output, including truncated image changes and clearer changes in source code compared to JSON.}
+    \label{fig:nbdiff}
+\end{figure}
+
+nbdime provides two mechansims for comparing notebooks. First is a command-line diff of notebooks
+via the \texttt{nbdiff} command, enabling use in a text-based terminal where developers spend much
+of their time. The key for diffing notebooks is that not all information in a notebook has equal
+importance, and not all of it is sensibly viewable as text, such as embedded images. The second
+important task of nbdime is displaying text content in a more legible form than the raw JSON. JSON
+includes additional structured markup, such as quotation marks, indentation, and escaped
+characters, which make the text difficult to read, and appear different from the text a user might
+see when they are editing the notebook. By recognizing the structure of the notebook, nbdime is
+able to make intelligent decisions about how to display text without additional markup and deemphasize information that is not viewable in a text-only environment.
+nbdime includes the structure and metadata with minimal markup, designed for humans to read.
+When images or other embedded data is seen, it is replaced by an indication that there is binary data that change, rather than the unintelligible difference between the binary data.
+
+\begin{figure}
+    \center
+    \includegraphics[width=10cm]{img/nbdiff-web}
+    \caption{nbdiff-web output showing side-by-side comparison of output images.}
+    \label{fig:nbdiff-web}
+\end{figure}
+
+nbdime also provides a web-based diff viewer via the \texttt{nbdiff-web} command. Notebooks are
+most commonly viewed and produced in a web-based environment, with the ability to view rendered
+images and HTML. The goal of the web-based viewer is to show the reviewer the actual changes in the
+notebook document, including changes in output images and rendered HTML.
+This allows the most natural experience of reviewing changes in a document, where the review environment is as similar to the interactive notebook environment as possible.
+
+% section diffing_notebooks (end)
+
+\section{merging notebooks} % (fold)
+\label{sec:merging_notebooks}
+
+\begin{figure}
+    \center
+    \includegraphics[width=8cm]{img/nbmerge}
+    \caption{The result of nbmerge with a conflict is still a valid notebook.}
+    \label{fig:nbmerge}
+\end{figure}
+
+One challenge for merging two notebooks using text is the structured JSON information, where
+text-based merge strategies can result in a syntactically invalid document. This is a problem for
+merging all structured documents, including source code in programming languages such as C, Python,
+or \LaTeX, but for tools where the editor is not plain-text, preventing the creation of these
+invalid documents. Another challenge for merging is the fact that some of the content in notebooks
```

Comment:
"This is a problem for merging all structured documents, including source code in programming languages such as C, Python, or \LaTeX, but for tools where the editor is not plain-text, preventing the creation of these invalid documents."

Check sentence (I do not understand it as it is, but I understand what it is intended to say).